src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let separatedString = '';
+    const lastIndex = this.length - 1;
+
+    for (let i = 0; i < this.length; i++) {
+      let temp = this[i];
+
+      if (this[i] === null || this[i] === undefined) {
+        temp = '';
+      }
+
+      if (i !== lastIndex) {
+        separatedString += temp + String(separator);
+      } else {
+        separatedString += temp;
+      }
+    }
```

Comment:
No need to reassign value, just add to result if it has value,
also add separator if i less then lastIndex
```suggestion
    for (let i = 0; i < this.length; i++) {
      const result = this[i];

      if (result !== null && result  !== undefined) {
        separatedString += result ;
      }

      if (i < lastIndex) {
        separatedString += separator;
      }
    }
```